src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,114 @@
 package core.basesyntax;
 
+import java.util.HashMap;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float DEFAULT_LOAD_FACTORY = 0.75f;
+    private static final float LOAD_FACTORY = DEFAULT_LOAD_FACTORY;
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int INCREASE_TABLE = 2;
+    private static final int HASH_SHIFT = 16;
+    private int initialCapacity = DEFAULT_CAPACITY;
+    private int threshold;
+    private Node<K, V>[] table;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public MyHashMap() {
+        this.table = (Node<K, V>[]) new Node[initialCapacity];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        threshold = (int) (initialCapacity * LOAD_FACTORY);
+        if (size >= threshold) {
+            resize();
+        }
+        int index = Math.abs(hash(key)) % table.length;
```

Comment:
Create informative variable and method names. Writing proper variable names can be a highly valuable investment to the quality of your code. Not only you and your teammates understand your code better, but it can also improve code sustainability in the future later on. When you go back to the same code base and re-read it over again, you should understand what is going on. Do not use abstract words like `string` or `array` as variable name. Do not use one-letter names. The name of the method should make it clear what it does.